### ISSUE TITLE
DOC failure while trying to follow "Debugging" section of the docs

### ISSUE THREAD

**Issue by adrinjalali** [numpy/numpy#26069]
I'm trying to debug something, while following https://numpy.org/devdocs/dev/development_environment.html#debugging

However, `spin gdb test.py` followed by `run` gives this error:

```
$ spin gdb /tmp/test.py
!! Could not load command `lldb` from file `spin.cmds.meson`.

Invoking `build` prior to invoking gdb:
$ /home/adrin/miniforge3/envs/numpy/bin/python3.11 vendored-meson/meson/meson.py compile -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /home/adrin/miniforge3/envs/numpy/bin/ninja -C /home/adrin/Projects/gh/me/numpy/build
ninja: Entering directory `/home/adrin/Projects/gh/me/numpy/build'
[1/1] Generating numpy/generate-version with a custom command
Saving version to numpy/version.py
$ /home/adrin/miniforge3/envs/numpy/bin/python3.11 vendored-meson/meson/meson.py install --only-changed -C build --destdir ../build-install
$ export PYTHONPATH="/home/adrin/Projects/gh/me/numpy/build-install/usr/lib/python3.11/site-packages"
$ gdb -ex 'set detach-on-fork on' --args /home/adrin/miniforge3/envs/numpy/bin/python3.11 /tmp/test.py
GNU gdb (GDB) 14.2
Copyright (C) 2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /home/adrin/miniforge3/envs/numpy/bin/python3.11...
(gdb) run
Starting program: /home/adrin/miniforge3/envs/numpy/bin/python3.11 /tmp/test.py

This GDB supports auto-downloading debuginfo from the following URLs:
  <https://debuginfod.archlinux.org>
Enable debuginfod for this session? (y or [n]) n
Debuginfod has been disabled.
To make this setting permanent, add 'set debuginfod enabled off' to .gdbinit.
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
+ /home/adrin/miniforge3/envs/numpy/bin/ninja
[Detaching after vfork from child process 134050]
Traceback (most recent call last):
  File "/tmp/test.py", line 1, in <module>
    import numpy as np
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1138, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1078, in _find_spec
  File "/home/adrin/miniforge3/envs/numpy/lib/python3.11/site-packages/_numpy_editable_loader.py", line 271, in find_spec
    tree = self.rebuild()
           ^^^^^^^^^^^^^^
  File "/home/adrin/miniforge3/envs/numpy/lib/python3.11/site-packages/_numpy_editable_loader.py", line 312, in rebuild
    subprocess.run(self._build_cmd, cwd=self._build_path, env=env, stdout=stdout, check=True)
  File "/home/adrin/miniforge3/envs/numpy/lib/python3.11/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adrin/miniforge3/envs/numpy/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/adrin/miniforge3/envs/numpy/lib/python3.11/subprocess.py", line 1953, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/home/adrin/Projects/gh/me/numpy/build/cp311'
[Inferior 1 (process 134047) exited with code 01]
```

I've tried this with a fresh build from the `main` branch. And the error message kinda makes sense since this is the contents of the `build` folder:

```
$ ll build
total 836
-rw-r--r-- 1 adrin adrin 596495 Mar 18 18:19 build.ninja
-rw-r--r-- 1 adrin adrin 237311 Mar 18 18:19 compile_commands.json
drwxr-xr-x 6 adrin adrin   4096 Mar 18 18:19 meson_cpu
drwxr-xr-x 2 adrin adrin   4096 Mar 18 18:19 meson-info
drwxr-xr-x 2 adrin adrin   4096 Mar 18 18:20 meson-logs
drwxr-xr-x 2 adrin adrin   4096 Mar 18 18:19 meson-private
drwxr-xr-x 6 adrin adrin   4096 Mar 18 18:19 numpy
```

and I've used this command to build:

```
CFLAGS="-O0 -g" CXXFLAGS="-O0 -g" spin build --clean -- -Dbuildtype=debug -Ddisable-optimization=true
```

**Comment by adrinjalali** [comment]
Or even simpler:

```py
$ python -c "import numpy"
+ /home/adrin/miniforge3/envs/numpy/bin/ninja
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1138, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1078, in _find_spec
  File "/home/adrin/miniforge3/envs/numpy/lib/python3.11/site-packages/_numpy_editable_loader.py", line 271, in find_spec
    tree = self.rebuild()
           ^^^^^^^^^^^^^^
  File "/home/adrin/miniforge3/envs/numpy/lib/python3.11/site-packages/_numpy_editable_loader.py", line 312, in rebuild
    subprocess.run(self._build_cmd, cwd=self._build_path, env=env, stdout=stdout, check=True)
  File "/home/adrin/miniforge3/envs/numpy/lib/python3.11/subprocess.py", line 548, in run
    with Popen(*popenargs, **kwargs) as process:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/adrin/miniforge3/envs/numpy/lib/python3.11/subprocess.py", line 1026, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/home/adrin/miniforge3/envs/numpy/lib/python3.11/subprocess.py", line 1953, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: '/home/adrin/Projects/gh/me/numpy/build/cp311'
```

**Comment by ngoldbaum** [comment]
I guess the docs need to clarify this, but the script can't be run from inside the numpy repository. Try doing something like this:

```bash
nathan at rous in ~/Documents/numpy on main
± cat ../numpy-experiments/test.py  
import numpy as np
x = np.arange(5)
np.empty_like(x)

nathan at rous in ~/Documents/numpy on main
± spin gdb ../numpy-experiments/test.py
Invoking `build` prior to invoking gdb:
$ /home/nathan/.pyenv/versions/3.11.3/bin/python3.11 vendored-meson/meson/meson.py compile -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /home/nathan/.pyenv/versions/3.11.3/bin/ninja -C /home/nathan/Documents/numpy/build
ninja: Entering directory `/home/nathan/Documents/numpy/build'
[1/1] Generating numpy/generate-version with a custom command
Saving version to numpy/version.py
$ /home/nathan/.pyenv/versions/3.11.3/bin/python3.11 vendored-meson/meson/meson.py install --only-changed -C build --destdir ../build-install
$ export PYTHONPATH="/home/nathan/Documents/numpy/build-install/usr/lib/python3.11/site-packages"
$ gdb -ex 'set detach-on-fork on' --args /home/nathan/.pyenv/versions/3.11.3/bin/python3.11 ../numpy-experiments/test.py
GNU gdb (Ubuntu 12.1-0ubuntu1~22.04) 12.1
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<https://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /home/nathan/.pyenv/versions/3.11.3/bin/python3.11...
(gdb) break array_empty_like
Function "array_empty_like" not defined.
Make breakpoint pending on future shared library load? (y or [n]) y
Breakpoint 1 (array_empty_like) pending.
(gdb) r
Starting program: /home/nathan/.pyenv/versions/3.11.3/bin/python3.11 ../numpy-experiments/test.py
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff35ff640 (LWP 329601)]
[New Thread 0x7ffff2dfe640 (LWP 329602)]
[New Thread 0x7fffe25fd640 (LWP 329603)]
[New Thread 0x7fffe1dfc640 (LWP 329604)]
[New Thread 0x7fffd95fb640 (LWP 329605)]
[New Thread 0x7fffc8dfa640 (LWP 329606)]
[New Thread 0x7fffc05f9640 (LWP 329607)]

Thread 1 "python3.11" hit Breakpoint 1, array_empty_like (__NPY_UNUSED_TAGGEDignored=<module at remote 0x7ffff6dc2070>, args=0x7ffff7fb0078, len_args=1, kwnames=0x0) at ../numpy/_core/src/multiarray/multiarraymodule.c:2029
2029	{

```

The docs should also probably mention `spin lldb`, which was added after those docs were written.

**Comment by adrinjalali** [comment]
My test script is indeed not in my numpy directory at all.

I figured the issue. It was because I had already installed with `pip install -e ....`. Had to do a `pip uninstall numpy` and then things work.

There's already somewhere the mention of not doing `pip` and `spin` together, so I did `git clean -xdf` before building with `spin` and since `spin build` docs says it builds and installs, I assumed it'll override the installed version, but it didn't.

So closing this one.
